### PR TITLE
Filtering with Mia's ML Script

### PR DESCRIPTION
**Pull request by tanzeelak**
## Types of changes
- [ ] Bugfix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [x] Refactor (change which changes the codebase without affecting its external behavior)
- [x] Non-breaking change (fix or feature that would causes existing functionality to work as expected)
- [ ] Breaking change (fix or feature that would cause existing functionality to **not** work as expected)

## Purpose
We currently store both Tumblr posts from the old blog and Keystone posts from the new Admin UI. Previously, we implemented filtering the tumblr posts by parsing the `tags` object from Tumblr. This involved Ishaan and I manually adding a corresponding tag to each of the tumblr posts. Since we want old tumblr posts to be incorporated with the new blog's filtering feature, we didn't want to manually add tags to the existing tumblr posts (734 posts). Instead, Mia used machine learning to sort the existing tumblr posts into categories. 

## Approach

Mia's script categorized all the existing tumblr posts into the relevant categories: "Show Reviews", "Music Reviews". She used the title and content to sort. To create the training set, she used the titles. Once she does that, she uses the training sets content and assesses similarities to the unlabeled articles content. This label is attached to original JSON post object as `topic`. She outputs a categorized blogposts object.

Existing tumblr posts were fetched and stored in the mongo database via a script run by `yarn fill-blog-db`. This is no longer necessary, as we will now be importing Mia's categorized blogposts object into mongo. 

I refactored tagging on the frontend for both Keystone and Tumblr. We used to filter based on the last value of the `tags` array in the post object. Now, we extract from `topic` in the post object. Keystone's post object previously also used a `tags` array. I updated the deployed version of Keystone post object to include a `category` object. Tumblr post's `topic` and Keystone post's `category` are treated the same. They are converted into filter names on both BlogPage.js and BlogPostPage.js, where they are eventually displayed. I had to update the existing filters to include all the new filters.

I also included Arjun's fix from [#316](https://github.com/uclaradio/uclaradio.com/pull/316). His works for the old filter names, but the new filter names have changed significantly enough that I included them here. 
"Previously, when selecting filters, dismissing the filter box would cause the checked boxes to reset to their default, unchecked state (the filters themselves were still active, of course). This was remedied with the addition of a checked attribute to the Input tags for the checkboxes, calling containsFilter to determine whether a filter is active or not."

## Testing

1. Delete existing mongo database
In one terminal, start mongod: `mongod`
In a second terminal, enter the mongo shell: `mongo`
Choose the uclaradio database: `use uclaradio`
Drop the blogposts collection: `db.blogposts.drop()`
`Ctrl+C` out of the mongo shell.
2. Import the newly generated posts that Mia has provided: 
Download [categorized-blogposts.json from Google Drive](https://drive.google.com/file/d/1-_BgSih7qMNPyroSKPOExeT_Moafh4d6/view?usp=sharing)
Move this file to the uclaradio repo directory.
Import the json file into mongo's blogposts collection. `mongoimport --jsonArray --db uclaradio --collection blogposts --file categorized-blogposts.json`
3. Run the website: `yarn dev`. Navigate to `localhost:3000/blog`. You should see that most of the posts have categories.
4. Press "Categories" bar and select "Show Reviews" , "Interviews", and "Sports".
5. Click on any of the posts to check that the category is consistent.

#### Blog Page
screen shot 2019-03-04 at 10 53 43 pm

### Blog Post Page
screen shot 2019-03-04 at 11 11 41 pm

## Checklist

- [x] My branch follows the branch naming scheme of UCLA Radio, and can merge into `master` without error.
- [ ] My code follows the code style of this project, and I have linted it to confirm this.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

## Link to Issue
[#289](https://github.com/uclaradio/uclaradio.com/issues/289)
[#315](https://github.com/uclaradio/uclaradio.com/issues/315)

### CODE REVIEW

File: client/react/frontpage/components/BlogPage.js [hunk 1]
```diff
@@ -169,31 +180,48 @@ const BlogPage = React.createClass({
         }
     }
   },
-  tagToType(tag) {
+  tagToCategory(tag) {
+    tag = tag.toLowerCase().replace(/ /g, '');
```
Comment (haejinjo) [inline client/react/frontpage/components/BlogPage.js]: nice. you _betta_ globalize!

Comment (haejinjo) [inline client/react/frontpage/components/BlogPage.js]: nice. you _betta_ globalize!

File: client/react/frontpage/components/BlogPage.js [hunk 2]
```diff
@@ -169,31 +180,48 @@ const BlogPage = React.createClass({
...
-        return 'CONCERT REVIEW';
-      case 'ArtistInterview':
-        return 'ARTIST INTERVIEW';
-      case 'Sports':
-        return 'SPORTS';
-      case 'FestivalReview':
-        return 'FESTIVAL REVIEW';
-      case 'Other':
-        return 'UCLA RADIO';
+      case 'concertreview':
```
Comment (haejinjo) [inline client/react/frontpage/components/BlogPage.js]: Are users the ones who tag these posts on the heroku entry tool? If so, we should be careful in the future about standardizing these tags.

Comment (tanzeelak) [reply client/react/frontpage/components/BlogPage.js]: In the Heroku entry tool, users will be able to select from categories, they can't directly input the category.

Comment (tanzeelak) [reply client/react/frontpage/components/BlogPage.js]: I have buncha cases for all the tags mia generated from the tumblr posts

**Pull request merged**